apps/dashboard/src/components/workflow-editor/workflow-actions.tsx

```diff
@@ -0,0 +1,74 @@
+import { useFetchIntegrations } from '@/hooks/use-fetch-integrations';
+import { ROUTES } from '@/utils/routes';
+import { ChannelTypeEnum, StepTypeEnum, WorkflowResponseDto } from '@novu/shared';
+import { RiArrowRightUpLine, RiTimeLine } from 'react-icons/ri';
+import { useNavigate } from 'react-router-dom';
+import { Badge, BadgeIcon } from '../primitives/badge';
+import { Button } from '../primitives/button';
+import { HoverCard, HoverCardContent, HoverCardTrigger } from '../primitives/hover-card';
+import { SidebarContent } from '../side-navigation/sidebar';
+import TruncatedText from '../truncated-text';
+
+interface WorkflowActionsProps {
+  workflow: WorkflowResponseDto;
+}
+
+export function WorkflowActions({ workflow }: WorkflowActionsProps) {
+  const { integrations } = useFetchIntegrations();
+  const navigate = useNavigate();
+
+  const hasInAppStep = workflow.steps.some((step) => step.type === StepTypeEnum.IN_APP);
+  const hasConnectedInAppIntegration = integrations?.some(
+    (integration) => integration.channel === ChannelTypeEnum.IN_APP && integration.connected
+  );
+
+  const showInAppActionRequired = hasInAppStep && !hasConnectedInAppIntegration;
+
+  const handleInboxSetup = () => {
+    navigate(ROUTES.INBOX_EMBED);
```

Comment:
we should apply the env filter when navigating to this page, otherwise it might say that you already integrated inbox

Reply:
Ooops good catch, I had a query param for this